packages/odf/modals/StorageClientDistributionModal/StorageClientResourceAttacherActions.ts

```diff
@@ -0,0 +1,79 @@
+import { useMemo } from 'react';
+import {
+  getDriver,
+  getProvisioner,
+  StorageClassModel,
+  StorageClassResourceKind,
+  VolumeSnapshotClassKind,
+  VolumeSnapshotClassModel,
+} from '@odf/shared';
+import { isCephDriver, isCephProvisioner } from '@odf/shared/utils';
+import {
+  Action,
+  K8sModel,
+  useModal,
+} from '@openshift-console/dynamic-plugin-sdk';
+import { LaunchModal } from '@openshift-console/dynamic-plugin-sdk/lib/app/modal-support/ModalProvider';
+import { StorageClientAttacherModal } from './StorageClientAttacherComponent';
+
+const isStorageClass = (
+  resource: StorageClassResourceKind | VolumeSnapshotClassKind
+): resource is StorageClassResourceKind =>
+  resource.kind === StorageClassModel.kind;
+
+const isVolumeSnapshotClass = (
+  resource: StorageClassResourceKind | VolumeSnapshotClassKind
+): resource is VolumeSnapshotClassKind =>
+  resource.kind === VolumeSnapshotClassModel.kind;
+
+export const useStorageClientAttacherAction = (
+  resource: StorageClassResourceKind | VolumeSnapshotClassKind
+) => {
+  const launchModal = useModal();
+  const actions = useMemo(() => {
+    const items = [];
+    if (
+      isStorageClass(resource) &&
+      isCephProvisioner(getProvisioner(resource))
+    ) {
+      items.push(
+        AttachResourceToStorageClientModal(
+          resource,
+          StorageClassModel,
+          launchModal
+        )
+      );
+    }
+    if (isVolumeSnapshotClass(resource) && isCephDriver(getDriver(resource))) {
+      items.push(
+        AttachResourceToStorageClientModal(
+          resource,
+          VolumeSnapshotClassModel,
+          launchModal
+        )
+      );
+    }
+    return items;
+  }, [resource, launchModal]);
+  return useMemo(() => [actions, true, undefined], [actions]);
```

Comment:
You can achieve same functionality with a single `useMemo` hook...
```suggestion
  const actions = useMemo(() => {
    const items = [];
    if (
      isStorageClass(resource) &&
      isCephProvisioner(getProvisioner(resource))
    ) {
      items.push(
        AttachResourceToStorageClientModal(
          resource,
          StorageClassModel,
          launchModal
        )
      );
    }
    if (isVolumeSnapshotClass(resource) && isCephDriver(getDriver(resource))) {
      items.push(
        AttachResourceToStorageClientModal(
          resource,
          VolumeSnapshotClassModel,
          launchModal
        )
      );
    }
    return [items, true, undefined];
  }, [resource, launchModal]);
  return actions;
```